packages/subproviders/src/subproviders/mnemonic_wallet_subprovider.ts

```diff
@@ -0,0 +1,123 @@
+    constructor(mnemonic: string, derivationPath: string = DEFAULT_DERIVATION_PATH) {
+        assert.isString('mnemonic', mnemonic);
+        super();
+        this._hdKey = HDNode.fromMasterSeed(bip39.mnemonicToSeed(mnemonic));
+        this._derivationPathIndex = 0;
+        this._derivationPath = derivationPath;
+    }
+    /**
+     * Retrieve the set derivation path
+     * @returns derivation path
+     */
+    public getPath(): string {
+        return this._derivationPath;
+    }
+    /**
+     * Set a desired derivation path when computing the available user addresses
+     * @param derivationPath The desired derivation path (e.g `44'/60'/0'`)
+     */
+    public setPath(derivationPath: string) {
+        this._derivationPath = derivationPath;
+    }
+    /**
+     * Set the final derivation path index. If a user wishes to sign a message with the
+     * 6th address in a derivation path, before calling `signPersonalMessageAsync`, you must
+     * call this method with pathIndex `6`.
+     * @param pathIndex Desired derivation path index
+     */
+    public setPathIndex(pathIndex: number) {
+        this._derivationPathIndex = pathIndex;
+    }
+    /**
+     * Retrieve the account associated with the supplied private key.
+     * This method is implicitly called when issuing a `eth_accounts` JSON RPC request
+     * via your providerEngine instance.
+     * @return An array of accounts
+     */
+    public async getAccountsAsync(numberOfAccounts: number = DEFAULT_NUM_ADDRESSES_TO_FETCH): Promise<string[]> {
+        const accounts: string[] = [];
+        for (let i = 0; i < numberOfAccounts; i++) {
+            const derivedHDNode = this._hdKey.derive(`m/${this._derivationPath}/${i + this._derivationPathIndex}`);
+            const derivedPublicKey = derivedHDNode.publicKey;
+            const shouldSanitizePublicKey = true;
+            const ethereumAddressUnprefixed = ethUtil
+                .publicToAddress(derivedPublicKey, shouldSanitizePublicKey)
+                .toString('hex');
+            const ethereumAddressPrefixed = ethUtil.addHexPrefix(ethereumAddressUnprefixed);
+            accounts.push(ethereumAddressPrefixed.toLowerCase());
+        }
+        return accounts;
+    }
+
+    /**
+     * Signs a transaction with the from account (if specificed in txParams) or the first account.
+     * If you've added this Subprovider to your  app's provider, you can simply send
+     * an `eth_sendTransaction` JSON RPC request, and * this method will be called auto-magically.
+     * If you are not using this via a ProviderEngine instance, you can call it directly.
+     * @param txParams Parameters of the transaction to sign
+     * @return Signed transaction hex string
+     */
+    public async signTransactionAsync(txParams: PartialTxParams): Promise<string> {
+        const accounts = await this.getAccountsAsync();
+        const hdKey = this._findHDKeyByPublicAddress(txParams.from || accounts[0]);
```

Comment:
Can't we just get this from the set `pathIndex` and `derivationPath`. We should probably just assert that `txParams.from` should be equal to the address pointed to from the `derivationPath` + `pathIndex`.

Reply:
Hmm I always viewed that index as an offset into the set of available addresses.

From a perspective of automated trading, I can see a use case of using multiple accounts at the same time, derived from the mnemonic seed, and the `from` in `txParams` indicates the account to use (remember this can be through provider engine, which supports multiple accounts). Rather than the user having to set the index then make request to `sendTransaction`. 

Requiring them to match might make sense in a dApp where the user has 'selected a single account', but if this was to be used in our tests (with multiple accounts off the mnemonic) then it limits the overall functionality of the Mnemonic Wallet Subprovider.